doc/development/autograph.rst

```diff
@@ -0,0 +1,743 @@
+which are updated across iterations of the loop must have a JAX compilable
+type (Booleans, Python numeric types, and JAX arrays).
+
+You can also utilize temporary variables within a while loop:
+
+>>> def f(x):
+...     while x < 5:
+...         c = "hi"
+...         x = x + 2 * len(c)
+...     return x
+>>> plxpr = make_plxpr(f)(0.0)
+>>> eval_jaxpr(plxpr.jaxpr, plxpr.consts, 4.4)
+[Array(8.4, dtype=float64, weak_type=True)]
+
+Temporary variables used inside a loop --- and that are **not** passed to a
+function within the loop --- do not have any type restrictions.
+
+A caveat regarding updating variables in a while loop is that it is not possible to
+update variables inside the loop test statement. For example, while the following
+works in standard Python:
+
+>>> def fn(limit):
+...     i = 0
+...     y = 0
+...     while (i := y) < limit:
+...         y += 1
+...     return i
+>>> fn(10)
+10
+
+any updates to the variables inside the while test function (in this case ``(i := y)``)
+will be ignored by AutoGraph:
+
+>>> plxpr = make_plxpr(fn)(0)
+>>> jax.core.eval_jaxpr(plxpr.jaxpr, plxpr.consts, 10)
+[0]
+
+Logical statements
+------------------
+
+AutoGraph in PennyLane currently does not provide support for capturing logical statements that involve dynamic variables --- that is,
+statements involving ``and``, ``not``, and ``or`` that return booleans.
+
+Debugging
+---------
+
+One useful tool in debugging issues is to examine the plxpr representation
+of the compiled function, in order to verify that AutoGraph is correctly capturing
+the control flow. For example, consider:
+
+.. code-block:: python
+
+    def f(x, n):
+
+        for i in range(n):
+
+            if x > 5:
+                y = x ** 2
+            else:
+                y = x ** 3
+
+            x = x + y
+
+        return x
+
+We can verify that the control flow is being correctly captured and
+converted by examining the plxpr representation of the compiled
+program:
+
+>>> make_plxpr(f)(0.0, 0)
+{ lambda ; a:f64[] b:i64[]. let
+    c:f64[] = for_loop[
+      args_slice=slice(0, None, None)
+      consts_slice=slice(0, 0, None)
+      jaxpr_body_fn={ lambda ; d:i64[] e:f64[]. let
+          f:bool[] = gt e 5.0
+          g:f64[] = cond[
+            args_slice=slice(4, None, None)
+            consts_slices=[slice(2, 3, None), slice(3, 4, None)]
+            jaxpr_branches=[{ lambda a:f64[]; . let b:f64[] = integer_pow[y=2] a in (b,) }, { lambda a:f64[]; . let b:f64[] = integer_pow[y=3] a in (b,) }]
+          ] f True e e
+          h:f64[] = add e g
+        in (h,) }
+    ] 0 b 1 a
+  in (c,) }
+
+Here, we can see the ``cond`` operation inside the for loop, and
+the two branches of the ``if`` statement represented by the ``jaxpr_branches``
+list.
+
+In addition, the function :func:`~.autograph_source` is provided,
+and allows you to view the converted Python code generated by AutoGraph:
+
+>>> def f(n):
```

Comment:
Can this code block be within backticks? Might solve the formatting issues

Reply:
Unfortunately that did not work, but I'll figure something out. Worst case we can remove the blank lines - a little less human-readable, but it should fix the rendering.